### DISCUSSION TITLE
Can't resolve 'react/jsx-runtime'

### DISCUSSION THREAD

**Discussion by dpgregory** [gilbarbara/react-joyride#955] [General]
Just over a week ago I started getting the following error when building a project containing react-joyride:

ERROR in ./node_modules/react-joyride/dist/index.mjs 1015:0-46
Module not found: Error: Can't resolve 'react/jsx-runtime' in '******\node_modules\react-joyride\dist'
Did you mean 'jsx-runtime.js'?
BREAKING CHANGE: The request 'react/jsx-runtime' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

I updated to the latest react-joyride version, 2.6.1, but still the same issue. The project is still on react 17.0.0. I reverted the code to a known state that used to build, but that no longer does, so appears to be something external to my project. I removed react-joyride from the project and all it's references and it then builds, so that indicates it is something to do with react-joyride.

Anyone else had the same thing start to happen and how did you solve it?

**Comment by gilbarbara** [comment]
Hey @dpgregory 

Try pinning the version to 2.5.5 (remove the `^` from the version in your package.json).

**Reply by dpgregory** [reply]
Hi @gilbarbara thanks for giving something for me to try, it is now working.

**Reply by rago4** [reply]
Hey, I know it's an old issue, but just wanted to ask - does this mean that versions above `2.5.5` require React v18 or higher?
